packages/ui/src/components/va-tree-view/VaTreeView.demo.vue

```diff
@@ -1,59 +1,352 @@
 <template>
   <VbDemo>
-    <VbCard>
-      <va-button @click="$refs.treeView.expand()">
-        EXPAND ALL
-      </va-button>
-      <va-button @click="$refs.treeView.collapse()">
-        COLLAPSE ALL
-      </va-button>
-
-      <va-tree-view ref="treeView">
-        <va-tree-category
-          label="Category text"
-          is-open
-        >
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-          <va-tree-category
-            label="Category text"
-            is-open
-          >
-            <va-tree-node>
-              Node text
-            </va-tree-node>
-          </va-tree-category>
-        </va-tree-category>
+    <VbCard title="Default">
+      <va-tree-view
+        :nodes="nodes"
+        node-key="id"
+      />
+    </VbCard>
+    <VbCard title="Icons">
+      <va-tree-view
+        :nodes="nodesWithIcons"
+        node-key="id"
+        expand-all
+      >
+        <template #icon="{icon}">
+          <va-icon
+            v-if="icon"
+            :name="icon"
+            size="20px"
+          />
+        </template>
+      </va-tree-view>
+    </VbCard>
+    <VbCard title="Selection and icons">
+      <va-radio
+        v-for="(option, index) in selectionTypeOptions"
+        :key="index"
+        v-model="selectionType"
+        :option="option"
+      />
+      <br />
+      <p>
+        {{ JSON.stringify(selectedNodes) }}
+      </p>
+      <br />
+      <va-tree-view
+        v-model="selectedNodes"
```

Comment:
<img width="325" alt="image" src="https://user-images.githubusercontent.com/5394573/183222591-2380e899-67a4-4016-a698-06e2a5e508c4.png">

That doesn't seem robust enough.

I.e., the way I see it, selecting all nodes of category should also select category, which it doesn't currently.
Yet, when I select category - items are getting selected.

Value also doesn't make sense from user perspective. I would say that categories should not be included.

Text for docs example could be improved a bit I think. You can use videogame genres as categories and titles as items for instance 😁.